src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -123,6 +126,20 @@ public ProjectState(LanguageServices languageServices, ProjectInfo projectInfo)
         _projectInfo = ClearAllDocumentsFromProjectInfo(projectInfoFixed);
 
         _lazyChecksums = AsyncLazy.Create(static (self, cancellationToken) => self.ComputeChecksumsAsync(cancellationToken), arg: this);
+        _lazyContentHashToDocumentId = AsyncLazy.Create(static (self, cancellationToken) => self.ComputeContentHashToDocumentIdAsync(cancellationToken), arg: this);
+    }
+
+    private async Task<Dictionary<ImmutableArray<byte>, DocumentId>> ComputeContentHashToDocumentIdAsync(CancellationToken cancellationToken)
+    {
+        var result = new Dictionary<ImmutableArray<byte>, DocumentId>(ImmutableArrayComparer<byte>.Instance);
```

Comment:
>Dictionary [](http://example.com/codeflow?start=25&length=10)

Isn't it possible for two files to contain the same content?

Reply:
doc'ed in followup.